### PR TITLE
Assign pull request to project during creation (#28227)

### PR DESCRIPTION
**Pull request by GiteaBot**
Backport #28227 by @denyskon

When creating a pull request, allow directly assigning it to a project, as it is already possible for issues.

After:

grafik

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Review by denyskon — CHANGES_REQUESTED** [review]
Failure seems related, will check soon...

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -1449,6 +1449,17 @@ func CompareAndPullRequestPost(ctx *context.Context) {
 		return
 	}
 
+	if projectID > 0 {
+		if !ctx.Repo.CanWrite(unit.TypeProjects) {
+			ctx.Error(http.StatusBadRequest, "user hasn't the permission to write to projects")
+			return
+		}
+		if err := issues_model.ChangeProjectAssign(ctx, pullIssue, ctx.Doer, projectID); err != nil {
```
Comment (delvh) [inline routers/web/repo/pull.go]: Probably
```suggestion
		if err := issues_model.ChangeProjectAssign(pullIssue, ctx.Doer, projectID); err != nil {
```
if I had to guess

**Review by delvh — APPROVED** [review]

File: routers/web/repo/pull.go (hunk 1 above)
Comment (denyskon) [reply routers/web/repo/pull.go]: Seems so.... can you apply the code suggestion? As it is a Giteabot branch, I cannot update it

**Review by lafriks — APPROVED** [review]

**Comment by delvh** [comment]
@GiteaBot ping

**Review by denyskon — APPROVED** [review]

**Pull request merged**